templates/prometheus.go

```diff
@@ -45,25 +50,68 @@ var PrometheusTemplate = promv1.Prometheus{
 				Port:      intstr.FromString("web"),
 			}},
 		},
-		Resources: utils.GetResourceRequirements("prometheus"),
-		// prometheus tls -- will be uncommented
-		// Web: &promv1.WebSpec{
-		// 	TLSConfig: &promv1.WebTLSConfig{
-		// 		KeySecret: corev1.SecretKeySelector{
-		// 			Key: "tls.key",
-		// 		},
-		// 		Cert: promv1.SecretOrConfigMap{
-		// 			Secret: &corev1.SecretKeySelector{
-		// 				Key: "tls.crt",
-		// 			},
-		// 		},
-		// 		ClientAuthType: fmt.Sprint(tls.VerifyClientCertIfGiven),
-		// 		ClientCA: promv1.SecretOrConfigMap{
-		// 			ConfigMap: &corev1.ConfigMapKeySelector{
-		// 				Key: "service-ca.crt",
-		// 			},
-		// 		},
-		// 	},
-		// },
+		Resources:   utils.GetResourceRequirements("prometheus"),
+		ListenLocal: true,
+		Containers: []corev1.Container{{
+			Name: "kube-rbac-proxy",
+			Args: []string{
+				fmt.Sprintf("--secure-listen-address=0.0.0.0:%d", KubeRBACProxyPortNumber),
+				"--upstream=http://127.0.0.1:9090/",
+				"--logtostderr=true",
+				"--v=10",
+				"--tls-cert-file=/etc/tls-secret/tls.crt",
+				"--tls-private-key-file=/etc/tls-secret/tls.key",
+				"--client-ca-file=/etc/tls-cert/service-ca.crt",
+				"--config-file=/etc/kube-rbac-proxy/config-file.json",
+			},
+			Ports: []corev1.ContainerPort{{
+				Name:          "https",
+				ContainerPort: KubeRBACProxyPortNumber,
+			}},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "serving-cert",
+					MountPath: "/etc/tls-secret",
+				},
+				{
+					Name:      "ca-cert",
+					MountPath: "/etc/tls-cert",
+				},
+				{
+					Name:      "kube-rbac-config",
+					MountPath: "/etc/kube-rbac-proxy",
+				},
+			},
+		}},
+		Volumes: []corev1.Volume{
+			{
+				Name: "serving-cert",
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: "managed-ocs-prom-svc-secret",
+					},
+				},
+			},
+			{
+				Name: "ca-cert",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: "prometheus-cert-bundle",
```

Comment:
I want to use the one created by us because this way we are not dependent on OCP versions as we already add annotations to support a wider range of OCP versions.